core/src/main/scala/org/apache/spark/util/AkkaUtils.scala

```diff
@@ -30,6 +32,48 @@ import org.apache.log4j.{Level, Logger}
 import org.apache.spark.{Logging, SecurityManager, SparkConf, SparkEnv, SparkException}
 
 /**
+ * Binds a timeout to a configuration property so that a thrown akka timeout exception can be
+ * traced back to the originating value.  The main constructor takes a generic timeout and
+ * description while the auxilary constructor uses a specific property defined in the
+ * configuration.
+ * @param timeout_duration timeout duration in milliseconds
+ * @param timeout_description description to be displayed in a timeout exception
+ */
+class ConfiguredTimeout(timeout_duration: FiniteDuration, timeout_description: String = null) {
```

Comment:
scala convention is camelCase, so `timeoutDuration` and `timeoutDescription`.  But in this case, I think `timeout` is a little redundant anyway, so maybe just `duration` and `description`.  I don't think you should have a `null` default for `description`